docs/topics/Rate_Limits.md

```diff
@@ -68,92 +68,23 @@ Note that the normal rate-limiting headers will be sent in this response. The ra
 ```
 < HTTP/1.1 429 TOO MANY REQUESTS
 < Content-Type: application/json
-< Retry-After: 6457
+< Retry-After: 65
 < X-RateLimit-Global: true
 {
   "message": "You are being rate limited.",
-  "retry_after": 6457,
+  "retry_after": 64.57,
   "global": true
 }
 ```
 
-## More Precise Rate Limit Resets
+## Invalid Request Limit	
 
-By default, the `X-RateLimit-Reset` and `X-RateLimit-Reset-After` returns a time rounded up to the nearest second. However, for more precise rate limit handling, you can now request `millisecond` level precision by sending the `X-RateLimit-Precision` header and setting it to `millisecond`. This means that you will get a more precise response, rounded to the nearest millisecond.
+IP addresses that make too many invalid HTTP requests are automatically and temporarily restricted from accessing the Discord API. Currently, this limit is **10,000 per 10 minutes**. An invalid request is one that results in **401**, **403**, or **429** statuses.	
 
-###### Example Responses
+All applications should make reasonable attempts to avoid making invalid requests. For example:	
 
-If the `X-RateLimit-Precision` header isn't set, it defaults to `second` precision.
+- **401** responses are avoided by providing a valid token in the authorization header when required and by stopping further requests after a token becomes invalid	
+- **403** responses are avoided by inspecting role or channel permissions and by not making requests that are restricted by such permissions	
```

Comment:
Is there any option to avoid these for DM channels?
My whole logging is spammed with DM failures, though everything else is catched.

Reply:
Not preemptively, no. If you have a large application that relies on DMs, you will probably want to track which channels give you a 403 and not make further requests to those. You can also consider self-imposing a global ratelimit on DMs.

Reply:
I know that it wasn't possible yet, but this feature is really needed, - so take it as a hint for a missing feature.

Reply:
Not that I know of. You can't tell if you have the ability to talk to someone without trying it first.

> My whole logging is spammed with DM failures

Are you bulk DMing people for this to happen or is it just a really big bot that responds to help commands in DMs or something, if you don't mind me asking?

Reply:
a failure to create a DM is a 400, not a 403, so it doesn't contribute toward this limit

Reply:
> I know that it wasn't possible yet, but this feature is really needed, - so take it as a hint for a missing feature.

You should open a separate issue for that.

Reply:
> Are you bulk DMing people for this to happen or is it just a really big bot that responds to help commands in DMs or something, if you don't mind me asking?

My bot is in about 550k servers. Trust me, I'm familar with the discord api and the (API) ToS.
Some features which send DM messages:
- Reminders
- Reaction-Roles (DM confirmation after role (un)assignment)
- Lottery Results
- Help Messages
- Vote Reminders
- Vote Messages (after a successful vote you'll get a DM confirmation, about 20k/day)

I think this list is already sufficient to show that my DM's are "fine".

Reply:
> Trust me, I'm familar with the discord api and the (API) ToS.
Wasn't ever implying that you weren't, so apologies if I gave that impression, I realise I probably worded that awfully, sorry about that!

Probably worth making a new issue like Sinister said though. The best solution for now if logging is just the issue (and not the concern of the 403 hard limit) would be to just handle that exception as an edge case until a better solution can be agreed if possible :)